src/System.Private.CoreLib/shared/System/Environment.cs

```diff
@@ -108,12 +107,22 @@ public static string GetFolderPath(SpecialFolder folder, SpecialFolderOption opt
             return GetFolderPathCore(folder, option);
         }
 
+        public static int CurrentManagedThreadId => Thread.CurrentThread.ManagedThreadId;
+
         public static bool Is64BitProcess => IntPtr.Size == 8;
 
         public static bool Is64BitOperatingSystem => Is64BitProcess || Is64BitOperatingSystemWhen32BitProcess;
 
         public static OperatingSystem OSVersion => s_osVersion.Value;
 
+#if !CORERT
+        public static string StackTrace
+        {
+            [MethodImpl(MethodImplOptions.NoInlining)] // Prevent inlining from affecting where the stacktrace starts
+            get => new StackTrace(true).ToString(System.Diagnostics.StackTrace.TraceFormat.Normal);
+        }
+#endif
```

Comment:
Does this really need to be shared?  It's basically one line.  Wouldn't it be cleaner to have that in the Mono-specific code?

Reply:
Why not? Is the if-def really that bad?